rest/config.go

```diff
@@ -1463,8 +1505,17 @@ func (sc *ServerContext) fetchDatabase(ctx context.Context, dbName string) (foun
 		// inherit properties the bootstrap config
 		cnf.CACertPath = sc.Config.Bootstrap.CACertPath
 
+		// We need to check for corruption in the database config (CC. CBG-3292). If the fetched config doesn't match the
+		// bucket name we got the config from we need to maker this db context as corrupt. Then remove the context and
+		// in memory representation on the server context.
 		bucketCopy := bucket
-		cnf.Bucket = &bucketCopy
+		if bucket != *cnf.Bucket {
+			sc.handleInvalidDatabaseConfig(ctx, bucket, cnf)
+			return true, fmt.Errorf("mismatch in peristed databse bucket name vs the actual bucketname. Please correct.")
+		} else {
+			// no corruption detected carry on as usual
+			cnf.Bucket = &bucketCopy
+		}
```

Comment:
```suggestion
		cnf.Bucket = &bucketCopy
```

we can drop the else statement here for readability (nit)